kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/IngressReconciler.java

```diff
@@ -0,0 +1,125 @@
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+import io.kroxylicious.kubernetes.api.common.ConditionBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngressBuilder;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Reconciles a {@link KafkaProxyIngress} by checking whether the {@link KafkaProxy}
+ * referenced by the {@code spec.proxyRef.name} actually exists, setting a
+ * {@link Condition.Type#ResolvedRefs} {@link Condition} accordingly.
+ */
+public class IngressReconciler implements
+        Reconciler<KafkaProxyIngress> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IngressReconciler.class);
+    public static final String PROXY_EVENT_SOURCE_NAME = "proxy";
+    private final Clock clock;
+
+    IngressReconciler(Clock clock) {
+        this.clock = clock;
+    }
+
+    @Override
+    public List<EventSource<?, KafkaProxyIngress>> prepareEventSources(EventSourceContext<KafkaProxyIngress> context) {
+        InformerEventSourceConfiguration<KafkaProxy> configuration = InformerEventSourceConfiguration.from(
+                KafkaProxy.class,
+                KafkaProxyIngress.class)
+                .withName(PROXY_EVENT_SOURCE_NAME)
+                .withPrimaryToSecondaryMapper((KafkaProxyIngress ingress) -> {
+                    return ResourcesUtil.localRefAsResourceId(ingress, ingress.getSpec().getProxyRef());
+                })
+                .withSecondaryToPrimaryMapper(proxy -> {
+                    return ResourcesUtil.findReferrers(context,
+                            proxy,
+                            KafkaProxyIngress.class,
+                            ingress -> ingress.getSpec().getProxyRef());
+                })
+                .build();
+        return List.of(new InformerEventSource<>(configuration, context));
+    }
+
+    @Override
+    public UpdateControl<KafkaProxyIngress> reconcile(
+                                                      KafkaProxyIngress ingress,
+                                                      Context<KafkaProxyIngress> context)
+            throws Exception {
+
+        var now = ZonedDateTime.ofInstant(clock.instant(), ZoneId.of("Z"));
+
+        ConditionBuilder conditionBuilder = newResolvedRefsCondition(ingress, now);
+
+        var proxyOpt = context.getSecondaryResource(KafkaProxy.class, PROXY_EVENT_SOURCE_NAME);
+        LOGGER.info("spec.proxyRef.name resolves to: {}", proxyOpt);
+
+        if (proxyOpt.isPresent()) {
+            conditionBuilder.withStatus(Condition.Status.TRUE);
```

Comment:
Up for discussion whether this proxy-exists case should be `ResolvedRefs: true` or something else.